daemon/cmd/daemon_main.go

```diff
@@ -980,6 +980,9 @@ func init() {
 	flags.Bool(option.ExternalClusterIPName, false, "Enable external access to ClusterIP services (default false)")
 	option.BindEnv(option.ExternalClusterIPName)
 
+	flags.Bool(option.EnableCiliumEndpointBatch, false, "If set to true, CiliumEndpointBatch feature is enabled and if any CiliumEndpoint resource are created, updated or deleted in the cluster, all those changes are passed in as CiliumEndpointBatch updates to cilium agents.")
+	option.BindEnv(option.EnableCiliumEndpointBatch)
+
```

Comment:
This flag looks that is being only used to create for the function that creates the CRD. That code is only used by cilium-operator so this flag should be configured in cilium-operator. `operator/option/config.go`

Reply:
I think this will be later used in agent (for when agent needs to watch for CEB instead of CEP).

Reply:
Thanks @Weil0ng for confirming.
Yes `EnableCiliumEndpointBatch` flag is used by both `cilium-operator` and `cilium-agent`.  Workflow  is enable this flag  in Cilium-config and CEB specific operator configs/flags in `operator/option/config.go`

Reply:
Where is `EnableCiliumEndpointBatch` used in the `agent`? In [1]? If yes, that code is only used by `cilium-operator`.

[1]
```golang
	if option.Config.EnableCiliumEndpointBatch {
		g.Go(func() error {
			return createCEBCRD(clientset)
		})
	}
```

Reply:
Cilium agent uses this `EnableCiliumEndpointBatch`  flag to determine either it needs to watch `CiliumEndpointBatch` or `CiliumEndpoint`. I have that code change in subsequent PRs. 

Just to give preview, below mentioned line is where we make that determination.
https://github.com/cilium/cilium/blob/master/pkg/k8s/watchers/watcher.go#L433-L436

Reply:
> I have specific changes in subsequent PRs.

that's the part I was missing.

Reply:
Devil's advocate: Do we really need this flag? Why wouldn't we just enable this by default?

Reply:
In small, medium size clusters the current  model of broadcasting CEP updates across k8s-worker nodes works well.  The current solution suits better for large clusters, in which we have >= 5000 worker nodes. If we enable current feature on small or medium sized clusters, there is not really much benefit, at the same time all CEP updates will delivered at more latency compared to current levels.